internal/provider/name_data_source.go

```diff
@@ -0,0 +1,110 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &NameDataSource{}
+
+func NewNameDataSource() datasource.DataSource {
+	return &NameDataSource{}
+}
+
+// NameDataSource defines the data source implementation.
+type NameDataSource struct {
+	popts ProviderOpts
+}
+
+type NameDataSourceModel struct {
+	Id           types.String `tfsdk:"id"`
+	Name         types.String `tfsdk:"name"`
+	Registry     types.String `tfsdk:"registry"`
+	Repo         types.String `tfsdk:"repo"`
+	Digest       types.String `tfsdk:"digest"`
+	Tag          types.String `tfsdk:"tag"`
+	RegistryRepo types.String `tfsdk:"registry_repo"`
+}
+
+// Metadata implements datasource.DataSource.
+func (*NameDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_name"
+}
+
+// Schema implements datasource.DataSource.
+func (d *NameDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Reference string data source",
+
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
```

Comment:
WDYT about requiring this to be a reference-by-digest for now, since that's the main use case we have ourselves, and we can always loosen the requirement later. We can document it and give it the digest validator, and the docs for `tag` get a bit easier for now since it'll _always_ hallucinate the tag. Maybe we can even name it `pseudo_tag`?

Reply:
done! I like `pseudo_tag`